### PR TITLE
Issue 1749 gh actions - add once a week dependency check

### PR DESCRIPTION
**Pull request by sierra-moxon**

### CODE REVIEW

File: .github/workflows/check-dependencies.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+name: Build and test linkml-runtime package dependencies
```
Comment (pkalita-lbl) [inline .github/workflows/check-dependencies.yaml:1]: I'm not sure this accurately expresses what this is doing. Maybe change this to something like "Build and test linkml-runtime with latest dependency versions"?

File: .github/workflows/check-dependencies.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  # Allows you to run this workflow manually from the Actions tab
+    types: trigger-run-check-dependencies
+
+jobs:
+  test:
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
```
Comment (pkalita-lbl) [inline .github/workflows/check-dependencies.yaml:16]: Off-topic but there's not need to test 3.7 anymore. It has been EOL [for a little while now](https://devguide.python.org/versions/).

Comment (sierra-moxon) [reply .github/workflows/check-dependencies.yaml]: removed from two github actions

File: .github/workflows/check-dependencies.yaml (hunk 1 above)
Comment (sierra-moxon) [reply .github/workflows/check-dependencies.yaml]: sure.

**Review by pkalita-lbl — APPROVED** [review]

**Pull request merged**